src/hooks/useSections.ts

```diff
@@ -41,19 +54,20 @@ export default function useSections(
   // Fetch Recommendations Results
   const recommendationsResults = useFetchRecommendationPod(cioClient, recommendationsSections);
 
-  const sectionResults = { ...autocompleteResults, ...recommendationsResults };
-
-  const activeSectionsWithData = getActiveSectionsWithData(activeSections, sectionResults);
-
+  // Add to active sections the results data and refs when autocomplete results or recommendation results fetched
   useEffect(() => {
-    setActiveSections(zeroStateActiveSections ? zeroStateSections : sections);
-  }, [query, sections, zeroStateSections, zeroStateActiveSections]);
+    const sectionsResults = { ...autocompleteResults, ...recommendationsResults };
+    setActiveSectionsWithData(
+      getActiveSectionsWithData(activeSections, sectionsResults, sectionsRefs)
+    );
+  }, [autocompleteResults, recommendationsResults, activeSections]);
 
+  // Reset sections
```

Comment:
why do we need this effect? I'm wondering why we need to manage `activeSections` as its own state here at all instead of deriving it from other state passed in to this hook...

Reply:
Removed